actionview/lib/action_view/helpers/tags/base.rb

```diff
@@ -79,35 +79,25 @@ def add_default_name_and_id_for_value(tag_value, options)
         end
 
         def add_default_name_and_id(options)
-          if options.has_key?("index")
-            options["name"] ||= options.fetch("name"){ tag_name_with_index(options["index"], options["multiple"]) }
-            options["id"] = options.fetch("id"){ tag_id_with_index(options["index"]) }
-            options.delete("index")
-          elsif defined?(@auto_index)
-            options["name"] ||= options.fetch("name"){ tag_name_with_index(@auto_index, options["multiple"]) }
-            options["id"] = options.fetch("id"){ tag_id_with_index(@auto_index) }
-          else
-            options["name"] ||= options.fetch("name"){ tag_name(options["multiple"]) }
-            options["id"] = options.fetch("id"){ tag_id }
+          index = name_and_id_index(options)
+          options["name"] = options.fetch("name"){ tag_name(options["multiple"], index) }
+          options["id"] = options.fetch("id"){ tag_id(index) }
+          if namespace = options.delete("namespace")
+            options['id'] = options['id'] ? "#{namespace}_#{options['id']}" : namespace
           end
-
-          options["id"] = [options.delete('namespace'), options["id"]].compact.join("_").presence
-        end
-
-        def tag_name(multiple = false)
-          "#{@object_name}[#{sanitized_method_name}]#{"[]" if multiple}"
         end
 
-        def tag_name_with_index(index, multiple = false)
-          "#{@object_name}[#{index}][#{sanitized_method_name}]#{"[]" if multiple}"
+        def tag_name(multiple = false, index = nil)
+          [
+            @object_name,
+            index && "[#{index}]",
+            "[#{sanitized_method_name}]",
+            multiple ? "[]" : ""
```

Comment:
`nil.to_s` returns an empty string, so you could write this as: `"[]" if multiple`.